DataDog/libdatadog
# [crashtracker] Enable dynamic updating of metadata and config at runtime

# What does this PR do?

Makes the `CrashtrackerMetadata` and `CrashtrackerConfiguration` dynamically updatable, rather than fixed at crashtracker startup.

# Motivation

Ruby allows the user to update tags on the fly, so we need to support that.

# Additional Notes

This will also be useful for a sidecar setup, where we would want to keep the metadata local until its time to upload.

# How to test the change?

Run the `test_crash` test in crashtracker/api, and notice that the extra tag is output in the crash report.

## For Reviewers
